src/pages/receipts/csvFileReceipt.js

```diff
@@ -49,22 +58,59 @@ const csvFileReceiptTemplate = async (username, auth, route) => {
                       </div>
                     </div>
                   </div>`
+  
   document.getElementById("contentBody").innerHTML = template;
   document.getElementById("navbarNavAltMarkup").innerHTML = nonUserNavBar(username);
 }
 
-const csvInTransitButtonSubmit = () => {
-  document.getElementById("createTransitCsv").addEventListener("click", async (e)=> {
+const getInTransitFileType = () => {
+  document.getElementById("createTransitFile").addEventListener("click", async (e) => {
     e.preventDefault();
-    showAnimation();
-    const response = await getAllBoxes(`bptl`);
-    hideAnimation();
-    const allBoxesShippedBySiteAndNotReceived = getRecentBoxesShippedBySiteNotReceived(response.data);
-    let modifiedTransitResults = updateInTransitMapping(allBoxesShippedBySiteAndNotReceived);
-    generateInTransitCSVData(modifiedTransitResults);
+    const modalHeaderEl = document.getElementById("modalHeader");
+    const modalBodyEl = document.getElementById("modalBody");
+    modalHeaderEl.innerHTML = `
+                              <h4>Select a format to download In Transit file</h4>
+                              <button type="button" class="close" data-dismiss="modal" aria-label="Close" id="closeModal">
+                              <span aria-hidden="true">&times;</span></button>`
+
+    modalBodyEl.innerHTML =  `<div class="row">
+                                <div class="col">
+                                      <form>
+                                        <div class="form-check">
+                                          <input class="form-check-input" type="radio" name="fileFormat" value="xlsx" id="xlsxCheck">
+                                          <label class="form-check-label" for="xlsxCheck">
+                                            .XLSX (better readability)
+                                          </label>
+                                        </div>
+                                        <div class="form-check">
+                                          <input class="form-check-input" type="radio" name="fileFormat" value="csv" id="csvCheck">
+                                          <label class="form-check-label" for="csvCheck">
+                                            .CSV (for BSI upload)
+                                          </label>
+                                        </div>
+                                      </form>
+                                </div>
+                            </div>`
+    confirmFileSelection();
   })
 }
 
+const confirmFileSelection = () => {
+  const radios = document.querySelectorAll('input[name="fileFormat"]');
+  radios.forEach(radio => {
+    radio.addEventListener('click', async (e) => {
+      const radioVal = radio.value;
```

Comment:
I like the ease of the user flow. The user clicks on either radio button and the file gets downloaded to their browser.